src/current/v24.1/known-limitations.md

```diff
@@ -6,699 +6,624 @@ keywords: gin, gin index, gin indexes, inverted index, inverted indexes, acceler
-- Variable shadowing (e.g., declaring a variable with the same name in an inner block) is not supported in PL/pgSQL. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/117508)
-- Syntax for accessing members of composite types without parentheses is not supported. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/114687)
-- The `STRICT` option for the PL/pgSQL `INTO` statement is not supported. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/107854)
-- `NOT NULL` variable declarations are not supported. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/105243)
-
-#### Cursor Functionality
-
-- Cursors opened in PL/pgSQL execute their queries on opening, affecting performance and resource usage. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/111479)
-- `OPEN FOR EXECUTE` is not supported for opening cursors. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/117744)
-
-#### Exception Handling
+The following sections describe newly identified limitations in CockroachDB {{ page.version.version }}.
+{% endcomment %}
 
-- PL/pgSQL exception blocks cannot catch [transaction retry errors]({% link {{ page.version.version }}/transaction-retry-error-reference.md %}). [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/111446)
-- `RAISE` statements cannot be annotated with names of schema objects related to the error (i.e., using `COLUMN`, `CONSTRAINT`, `DATATYPE`, `TABLE`, or `SCHEMA`). [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/106237)
-- `RAISE` statements message the client directly, and do not produce log output. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/117750)
-- `ASSERT` debugging checks are not supported. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/117744)
+## Existing limitations
 
-### Limitations in User-Defined Functions (UDFs) and Stored Procedures
+The following sections describe existing limitations in CockroachDB releases prior to {{ page.version.version }}.
 
-- Transactions cannot be run within stored procedures. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/115294)
-- UDFs and stored procedures cannot call other UDFs or stored procedures from within their bodies. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/88198)
-- DDL statements (e.g., `CREATE TABLE`, `CREATE INDEX`) are not allowed within UDFs or procedures. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/110080)
-- UDF and stored procedure definitions do not support `OUT` and `INOUT` argument modes. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/100405)
-- The `setval` function cannot be resolved when used inside UDF bodies. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/110860)
+### SQL statements
```

Comment:
What do you think about this kind of structure?

- Database (incorporates SQL, Transactions, Schema and Indexes, Data Types)
- Deployment and Operations (incorporates Security (? I could see Security being a top level section) and - Performance Optimization)
- Disaster Recovery (incorporates Change Data Capture)

Once there are new limitations, it's possible the right-hand TOC will start to look very busy, because "New limitations" will eventually gain many of the sections from the Existing Limitations section. It seems almost like a table or something would be better, with a range of versions that have the limitation, instead of the current sections. I'm not sure that I care whether a limitation is new or not when I'm evaluating whether to install / upgrade to a specific version. WDYT?

Reply:
>     * Database (incorporates SQL, Transactions, Schema and Indexes, Data Types)

This is too broad IMO. I could see moving 1) Transactions into SQL (although the topics on isolation levels would feel out of place next to strictly SQL syntax topics) and 2) Data Types into Schema and Indexes. @shannonbradshaw, do you have an opinion on this?

>     * Deployment and Operations (incorporates Security (? I could see Security being a top level section) and - Performance Optimization)

Same as above; I think we have enough optimizer-specific limitations that they merit a separate subheading. Those limitations are not really affected by how you operate CRDB; they simply exist in the optimizer (so may affect the SQL you decide to use). 

>     * Disaster Recovery (incorporates Change Data Capture)

I think CDC has broader applications than just DR, hence its own category in our docs IA.

> Once there are new limitations, it's possible the right-hand TOC will start to look very busy, because "New limitations" will eventually gain many of the sections from the Existing Limitations section. It seems almost like a table or something would be better, with a range of versions that have the limitation, instead of the current sections. I'm not sure that I care whether a limitation is new or not when I'm evaluating whether to install / upgrade to a specific version. WDYT?

I haven't spelled out my plan yet, but carrying over the sections from "Existing" is not in my plan. My current thought is that new limitations should be categorized however the writer prefers -- most likely by feature name. This makes it easier to cross-reference the new limitations against the GA new feature list. Then for the next version of this doc, the person that moves the New limitations under the Existing heading (presumably a docs mgr?) will need to distribute them appropriately into the appropriate sections and/or includes. It's frankly more organizational work but potentially worth the effort. @shannonbradshaw curious about your thoughts here.